File: problems/2515-shortest-distance-to-target-string-in-a-circular-array/solution.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        min_dist = float('inf')
        found = False
        for i in range(n):
            if words[i] == target:
                found = True
                dist1 = abs(i - startIndex)
                dist2 = n - dist1
                min_dist = min(min_dist, min(dist1, dist2))
        if not found:
            return -1
        else:
            return min_dist

File: problems/2515-shortest-distance-to-target-string-in-a-circular-array/solution.cpp
class Solution {
public:
    int closestTarget(vector<string>& words, string target, int startIndex) {
        int n = words.size();
        int minDist = INT_MAX;
        bool found = false;

        for (int i = 0; i < n; ++i) {
            if (words[i] == target) {
                found = true;
                int dist1 = abs(i - startIndex);
                int dist2 = n - dist1;
                minDist = min(minDist, min(dist1, dist2));
            }
        }

        if (!found) {
            return -1;
        }

        return minDist;
    }
};